charts/kyverno/templates/helm-pre-delete-hook.yaml

```
@@ -0,0 +1,59 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+  creationTimestamp: null
+  name: kyverno-helm-hook-role
+rules:
+  - apiGroups:
+      - admissionregistration.k8s.io
+    resources:
+      - validatingwebhookconfigurations
+    verbs:
+      - delete
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+  name: kyverno-helm-hook-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: kyverno-helm-hook-role
+subjects:
+  - kind: ServiceAccount
+    name: kyverno-demo
+    namespace: kyverno
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ template "kyverno.fullname" . }}-hook-pre-delete
+  labels: {{ include "kyverno.labels" . | nindent 4 }}
+    app: kyverno
+  annotations:
+    "helm.sh/hook": pre-delete
+    "helm.sh/hook-delete-policy": hook-succeeded,hook-failed
+spec:
+  template:
+    spec:
+      serviceAccount: {{ template "kyverno.serviceAccountName" . }}
+      containers:
+        - name: kubectl
+          image: 'bitnami/kubectl:latest'
+          command:
+            - sh
+            - '-c'
+            - >-
+              kubectl delete validatingwebhookconfiguration kyverno-policy-validating-webhook-cfg
+              kubectl delete validatingwebhookconfiguration kyverno-resource-validating-webhook-cfg
+              kubectl delete mutatingwebhookconfiguration kyverno-policy-mutating-webhook-cfg
+              kubectl delete mutatingwebhookconfiguration kyverno-resource-mutating-webhook-cfg
+              kubectl delete mutatingwebhookconfiguration kyverno-verify-mutating-webhook-cfg
```

I think these might get evaluated as one massive command which won't work
```suggestion
            - >-
              kubectl delete validatingwebhookconfiguration kyverno-policy-validating-webhook-cfg ;
              kubectl delete validatingwebhookconfiguration kyverno-resource-validating-webhook-cfg ;
              kubectl delete mutatingwebhookconfiguration kyverno-policy-mutating-webhook-cfg ; 
              kubectl delete mutatingwebhookconfiguration kyverno-resource-mutating-webhook-cfg ;
              kubectl delete mutatingwebhookconfiguration kyverno-verify-mutating-webhook-cfg
```

---

right!